tests/Doctrine/Tests/DBAL/Driver/StatementIteratorTest.php

```diff
@@ -2,39 +2,91 @@
 
 namespace Doctrine\Tests\DBAL\Driver;
 
+use Doctrine\DBAL\Driver\IBMDB2\DB2Statement;
+use Doctrine\DBAL\Driver\Mysqli\MysqliStatement;
+use Doctrine\DBAL\Driver\OCI8\OCI8Statement;
+use Doctrine\DBAL\Driver\SQLAnywhere\SQLAnywhereStatement;
+use Doctrine\DBAL\Driver\SQLSrv\SQLSrvStatement;
 use Doctrine\DBAL\Driver\Statement;
 use Doctrine\DBAL\Driver\StatementIterator;
+use Doctrine\DBAL\Portability\Statement as PortabilityStatement;
+use PHPUnit\Framework\MockObject\MockObject;
+use function extension_loaded;
 
 class StatementIteratorTest extends \Doctrine\Tests\DbalTestCase
 {
-    public function testGettingIteratorDoesNotCallFetch()
+    /**
+     * @dataProvider statementProvider()
+     */
+    public function testGettingIteratorDoesNotCallFetch(string $class)
     {
-        $stmt = $this->createMock(Statement::class);
+        $stmt = $this->createPartialMock($class, ['fetch', 'fetchAll', 'fetchColumn']);
         $stmt->expects($this->never())->method('fetch');
         $stmt->expects($this->never())->method('fetchAll');
         $stmt->expects($this->never())->method('fetchColumn');
 
+        $stmt->getIterator();
+    }
+
+    public function testIteratorIterationCallsFetchOncePerStep()
+    {
+        $stmt = $this->createMock(Statement::class);
+        $this->configureStatement($stmt, $calls);
+
         $stmtIterator = new StatementIterator($stmt);
-        $stmtIterator->getIterator();
+
+        $this->assertIterationCallsFetchOncePerStep($stmtIterator, $calls);
     }
 
-    public function testIterationCallsFetchOncePerStep()
+    /**
+     * @dataProvider statementProvider()
+     */
+    public function testStatementIterationCallsFetchOncePerStep(string $class)
+    {
+        $stmt = $this->createPartialMock($class, ['fetch']);
+        $this->configureStatement($stmt, $calls);
+        $this->assertIterationCallsFetchOncePerStep($stmt, $calls);
+    }
+
+    private function configureStatement(MockObject $stmt, &$calls) : void
     {
         $values = ['foo', '', 'bar', '0', 'baz', 0, 'qux', null, 'quz', false, 'impossible'];
         $calls = 0;
 
-        $stmt = $this->createMock(Statement::class);
         $stmt->expects($this->exactly(10))
             ->method('fetch')
             ->willReturnCallback(function() use ($values, &$calls) {
                 $value = $values[$calls];
                 $calls++;
+
                 return $value;
             });
+    }
 
-        $stmtIterator = new StatementIterator($stmt);
-        foreach ($stmtIterator as $i => $_) {
+    private function assertIterationCallsFetchOncePerStep($iterator, &$calls) : void
+    {
+        foreach ($iterator as $i => $_) {
             $this->assertEquals($i + 1, $calls);
         }
     }
+
+    public static function statementProvider()
+    {
+        if (extension_loaded('ibm_db2')) {
```

Comment:
Unlike `MysqliStatement` and `SQLAnywhereStatement`, `DB2Statement` and other driver statement class declarations depend on corresponding extension symbols, so they only can be loaded if extensions are loaded.